libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -313,6 +314,11 @@ string IRGenerator::dispatchRoutine(ContractDefinition const& _contract)
 	}
 	else
 		t("fallback", "revert(0, 0)");
+	if (FunctionDefinition const* etherReceiver = _contract.etherReceiverFunction())
+		// TODO: should this revert for iszero(callvalue())?
```

Comment:
I guess yes?
@ekpyron @chriseth

Reply:
Ah no, confused stuff

Reply:
Yeah, decision was allowing zero ether and zero data for receive.